source/getting-started/core-concepts.md

```diff
@@ -1,58 +1,136 @@
-user clicks the upvote button, increase the `vote` property's value by 1_.
+consist of two parts:
 
-## Controllers
+* A template, for example `app/templates/components/welcome-message.hbs`
+* Optionally a source file written in JavaScript, for example
+  `app/components/welcome-message.js`.
+* **Components must contain a - in their name**. More about that later.
+
+Components differ from helpers in that they can output not only a value, but
+also manage DOM. An important property of components is that they are **isolated**.
+For example, a component's DOM is called "shadow" DOM.
+
+In this example, a component's template adds a div around some other DOM:
 
-Controllers are very much like components, so much so that in future versions of
-Ember, controllers will be replaced entirely with components. At the moment,
-components cannot be routed to (see below), but when this changes, it will be
-recommended to replace all controllers with components.
+```handlebars
+{{! app/templates/components/my-component.hbs }}
+<div class="inner">
+  {{yield}}
+</div>
+```
+
+```handlebars
+<span>Hello</span>
+<div class="outer">
+  {{#my-component}}
+    Hello there!
+  {{/my-component}}
+</div>
+```
+
+Because components deal with DOM, they are also responsible for maintaining
+the bulk of an application's state. Additionally they handle events like
+clicks, keyUp, and other more semtantic actions.
+
+## The Router
+
+The router maps a URL to a template. For example, when a user visits the `/posts`
+URL, the router might load the `posts` route. The router can also load
+nested routes. For example, if a blogging app had a list of blog posts on the
+left of the screen and showed the current blog post on the right, we'd say
+that the `posts/show` route was nested inside the `posts` route.
+
+Perhaps the most important thing to remember about Ember is that **the URL drives
+the application state**. URLs determines what template and route to load, and
+that in turn determines what model must be fetched.
 
 ## Models
 
-Models represent _persistent state_. For example, a blog application would want
-to save the content of a blog post when a user publishes it, and so the blog
-post would have a model defining it, perhaps called the `Post` model. A model
+Models represent persistent state. For example, a blog application would
+persist the content of a blog post when a user publishes it. A blog post model
+would represent the state of that persisted content (such as the post itself,
+the title, and the author).
+
+A model
 typically persists information to a server, although models can be configured to
-save to anywhere else, such as the browser's Local Storage.
+save to anywhere. For example to the localStorage API, or IndexDB.
 
-## Routes
+When visiting a URL, Ember attempts to be smart about what models should be
+loaded. For example, if you visit the `/posts` URL an index of all existing
+`Post` models will be fetched.
+
+## Controllers
```

Comment:
I do think this text does a great job of illustrating the use for controllers, but I'm not sure it belongs on the page giving the 10,000 foot view of Ember. That's why I'm inclined to omit controllers from this page, but move this text to `controllers/index`.